### PR TITLE
refine servicebus session message handler

### PR DESCRIPTION
**Pull request by neuqlz**
## Description
refine servicebus session message handler

### CODE REVIEW

File: spring-integration-azure/spring-integration-servicebus/src/main/java/com/microsoft/azure/spring/integration/servicebus/queue/ServiceBusQueueTemplate.java [hunk 1]
```diff
@@ -142,19 +144,22 @@ protected String buildCheckpointSuccessMessage(Message<?> message) {
...
 
+            Message<U> message = messageConverter.toMessage(serviceBusMessage,
+                    new MessageHeaders(headers), payloadType);
+            consumer.accept(message);
+
+            if (checkpointConfig.getCheckpointMode() == CheckpointMode.RECORD) {
+                return checkpointer.success().whenComplete((v, t) -> super.checkpointHandler(message, t));
+            }
             return CompletableFuture.completedFuture(null);
         }
```
Comment (samukce) [inline spring-integration-azure/spring-integration-servicebus/src/main/java/com/microsoft/azure/spring/integration/servicebus/queue/ServiceBusQueueTemplate.java]: Any chance we have unit tests for this changes?

**Comment by saragluna** [comment]
@neuqlz 
What's the context of this PR?

File: spring-integration-azure/spring-integration-servicebus/src/main/java/com/microsoft/azure/spring/integration/servicebus/queue/ServiceBusQueueTemplate.java (hunk 1 above)
Comment (neuqlz) [reply spring-integration-azure/spring-integration-servicebus/src/main/java/com/microsoft/azure/spring/integration/servicebus/queue/ServiceBusQueueTemplate.java]: yep, I will add unit tests for it later.

**Comment by neuqlz** [comment]
> @neuqlz
> What's the context of this PR?

It is related to this issue #594  and this PR #596. 

I think there is something to refine, so I create this PR to do it.

**Pull request merged**